components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -356,9 +364,82 @@ public void updateRoleName(int tenantId, String oldRoleName, String newRoleName)
             }
         } catch (SQLException e) {
             log.error("Error when executing the SQL : " + SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE);
-            throw new IdentitySCIMException("Error when reading the SCIM Group information from the " +
-                    "persistence store.", e);
+            throw new IdentitySCIMException("Error when reading the SCIM Group information from the persistence store.",
+                    e);
+        }
+        return roleList.toArray(new String[roleList.size()]);
+    }
+
+    /**
+     * List the groups created from SCIM with a attribute filter and search regex.
+     *
+     * @param searchAttributeName  Search attribute name.
+     * @param searchAttributeValue Search attribute value.
+     * @param tenantId             Tenant ID.
+     * @param domainName           Domain that needs to be filtered.
+     * @return List of SCIM groups.
+     * @throws IdentitySCIMException IdentitySCIMException when reading the SCIM Group information.
+     */
+    public String[] getGroupNameList(String searchAttributeName, String searchAttributeValue, Integer tenantId,
+            String domainName) throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+        String sqlQuery;
+
+        // Resolve sql query for filtering.
+        if (StringUtils.isNotEmpty(domainName)) {
+            // if the domain is given, domain needs to be searched in ROLE_NAME column as well.
+            sqlQuery = SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE_AND_ROLE_NAME;
+        } else {
+            sqlQuery = SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE;
+        }
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlQuery)) {
+                prepStmt.setInt(1, tenantId);
+                prepStmt.setString(2, searchAttributeName);
+                prepStmt.setString(3, searchAttributeValue);
+
+                // Append SQL_FILTERING_DELIMITER to ROLE_NAME param to filter in a given domain.
+                if (StringUtils.isNotEmpty(domainName)) {
+                    prepStmt.setString(4, domainName.toUpperCase() + "%");
+                }
+                try (ResultSet rSet = prepStmt.executeQuery()) {
+                    while (rSet.next()) {
+                        String roleName = rSet.getString(1);
+                        if (StringUtils.isNotEmpty(roleName)) {
+                            if (roleName.contains(CarbonConstants.DOMAIN_SEPARATOR)) {
```

Comment:
This condition is already handled with `removePrimaryDomainName` method. You can do `roleList.add(removePrimaryDomainName(roleName))` without this.

Reply:
Fixed: 497cb3c10e705b08f73a61c5004190d023109795